ntoskrnl/lpc/port.c

```diff
@@ -288,8 +287,49 @@ NtQueryInformationPort(IN HANDLE PortHandle,
                        IN ULONG PortInformationLength,
                        OUT PULONG ReturnLength)
 {
-    UNIMPLEMENTED;
-    return STATUS_NOT_IMPLEMENTED;
+    KPROCESSOR_MODE PreviousMode;
+    NTSTATUS Status;
+    PLPCP_PORT_OBJECT PortObject;
+
+    PAGED_CODE();
+
+    UNREFERENCED_PARAMETER(PortInformationClass);
+
+    PreviousMode = KeGetPreviousMode();
+    if (PreviousMode != KernelMode)
+    {
+        _SEH2_TRY
+        {
+            ProbeForWrite(PortInformation, PortInformationLength, sizeof(ULONG));
+            if (ARGUMENT_PRESENT(ReturnLength))
+            {
+                ProbeForWriteUlong(ReturnLength);
+            }
+        }
+        _SEH2_EXCEPT(EXCEPTION_EXECUTE_HANDLER) {
+            _SEH2_YIELD(return _SEH2_GetExceptionCode());
+        }
+        _SEH2_END;
+    }
+
+    if (ARGUMENT_PRESENT(PortHandle))
+    {
+        Status = ObReferenceObjectByHandle(PortHandle, GENERIC_READ, LpcPortObjectType, PreviousMode, (PVOID*)&PortObject, NULL);
```

Comment:
Kernel code style likes to have each argument of functions that take a ton of arguments (like this one), on their own lines.

Reply:
Fix.